src/core/src/core_logic/PatchInstaller.py

```diff
@@ -42,6 +43,10 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.last_still_needed_package_versions = None
         self.progress_template = "[Time available: {0} | A: {1}, S: {2}, F: {3} | D: {4}]\t {5}"
 
+        self.attempted_parent_update_count = 0
+        self.successful_parent_update_count = 0
+        self.failed_parent_update_count = 0
```

Comment:
Why are these scoped at this level again? Not immediately obvious due to size of file.

Reply:
These variables were initially being used only in the method install_updates and hence they were local variables but now they are being used in install_patches_in_batches also so made them global variables.